kiwi/tasks/image_resize.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2015 SUSE Linux GmbH.  All rights reserved.
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+usage: kiwi image resize -h | --help
+       kiwi image resize --target-dir=<directory> --size=<size>
+           [--root=<directory>]
+       kiwi image resize help
+
+commands:
+    resize
+        for disk based images, allow to resize the image to a new
+        disk geometry. The additional space is free and not in use
+        by the image. In order to make use of the additional free
+        space a repartition process is required like it is provided
+        by kiwi's oem boot code. Therefore the resize operation is
+        useful for oem image builds most of the time
+
+options:
+    --root=<directory>
+        the path to the root directory, if not specified kiwi
+        searches the root directory in build/image-root below
+        the specified target directory
+
+    --size=<size>
+        new size of the image. The value is either a size in bytes
+        or can be specified with m=MB or g=GB. Example: 20g
+
+    --target-dir=<directory>
+        the target directory to expect image build results
+"""
+import re
+import math
+
+# project
+from .base import CliTask
+from ..help import Help
+from ..logger import log
+from ..storage.subformat import DiskFormat
+
+from ..exceptions import (
+    KiwiImageResizeError
+)
+
+
+class ImageResizeTask(CliTask):
+    """
+    Implements resizing of disk images and their disk format
+
+    Attributes
+
+    * :attr:`manual`
+        Instance of Help
+    """
+    def process(self):
+        """
+        reformats raw disk image and its format to a new disk
+        geometry using the qemu tool chain
+        """
+        self.manual = Help()
+        if self._help():
+            return
```

Comment:
Meh. Why call the whole thing from man, instead of just:
```python
if self.command_args.get(''help'):
    return None
```
BTW "explicit is better than implicit" (`import this`), thus `return None` is better than just `return`.